<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Schuyler D. smith" />

<meta name="date" content="2019-07-23" />

<title>Soil Column ARG Dispersal Workflow</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Soil Column ARG Dispersal Workflow</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Soil Column ARG Dispersal Workflow</h1>
<h4 class="author">Schuyler D. smith</h4>
<h4 class="date">2019-07-23</h4>

</div>


<p><br> <br></p>
<div id="required-pacakges" class="section level1">
<h1>Required Pacakges</h1>
<pre class="r"><code>library(data.table)</code></pre>
<pre><code>## data.table 1.12.2 using 4 threads (see ?getDTthreads).  Latest news: r-datatable.com</code></pre>
<pre class="r"><code>library(phyloseq)
library(phylosmith)
library(ggplot2)</code></pre>
<pre><code>## Find out what&#39;s changed in ggplot2 at
## https://github.com/tidyverse/ggplot2/releases.</code></pre>
<pre class="r"><code>library(vegan)</code></pre>
<pre><code>## Loading required package: permute</code></pre>
<pre><code>## Loading required package: lattice</code></pre>
<pre><code>## This is vegan 2.5-5</code></pre>
<pre class="r"><code>library(schuylR)
library(gridExtra)</code></pre>
</div>
<div id="data-setup" class="section level1">
<h1>Data Setup</h1>
<pre class="r"><code>phylo_object &lt;- readRDS(&quot;../data/soil_column_ARGs.RDS&quot;); phylo_object</code></pre>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 320 taxa and 56 samples ]
## sample_data() Sample Data:       [ 56 samples by 8 sample variables ]
## tax_table()   Taxonomy Table:    [ 320 taxa by 3 taxonomic ranks ]</code></pre>
<pre class="r"><code>phylo_object &lt;- phylosmith::merge_treatments(phylo_object, &#39;Matrix&#39;, &#39;Treatment&#39;)
treatments &lt;- &#39;Matrix_Treatment&#39;
levels(phylo_object@sam_data$Matrix_Treatment) &lt;- c(&#39;Effluent&#39;, &#39;Manure&#39;, &#39;Soil&#39;, &#39;Manure-Treated Effluent&#39;, NA, &#39;Manure-Treated Soil&#39;)
phylo_object &lt;- phylosmith::set_treatment_levels(phylo_object, treatment = &#39;Matrix_Treatment&#39;, 
                order = c(&#39;Manure&#39;, &#39;Soil&#39;, &#39;Effluent&#39;, &#39;Manure-Treated Soil&#39;, &#39;Manure-Treated Effluent&#39;))

metadata &lt;- fread(&#39;../data/metadata.tsv&#39;)
gsub(&#39;SC.*_A&#39;, &#39;A&#39;, metadata$`Sample ID`) == sample_names(phylo_object)</code></pre>
<pre><code>##  [1] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
## [15] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
## [29] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
## [43] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE</code></pre>
<pre class="r"><code>sample_names(phylo_object) &lt;- paste(sapply(strsplit(metadata$`Sample ID`, &#39;_&#39;),  `[`, 1), metadata$Day, sep = &#39;_Day_&#39;)
phylo_object</code></pre>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 320 taxa and 56 samples ]
## sample_data() Sample Data:       [ 56 samples by 9 sample variables ]
## tax_table()   Taxonomy Table:    [ 320 taxa by 3 taxonomic ranks ]</code></pre>
</div>
<div id="manure-resistome-vs-soil-resistome" class="section level1">
<h1>Manure Resistome vs Soil Resistome</h1>
<pre class="r"><code>manure</code></pre>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 173 taxa and 4 samples ]
## sample_data() Sample Data:       [ 4 samples by 9 sample variables ]
## tax_table()   Taxonomy Table:    [ 173 taxa by 5 taxonomic ranks ]</code></pre>
<pre class="r"><code>soil</code></pre>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 176 taxa and 16 samples ]
## sample_data() Sample Data:       [ 16 samples by 9 sample variables ]
## tax_table()   Taxonomy Table:    [ 176 taxa by 5 taxonomic ranks ]</code></pre>
<div id="unique-genes-for-each-resistome" class="section level2">
<h2>Unique Genes for Each Resistome</h2>
<pre class="r"><code>unique_control_genes &lt;- unique_taxa(phylo_object, treatments, c(&#39;Manure&#39;, &#39;Soil&#39;, &#39;Effluent&#39;)); sapply(unique_control_genes, length)</code></pre>
<pre><code>##   Manure     Soil Effluent 
##       88        8       54</code></pre>
<pre class="r"><code>data.frame(taxa_proportions(manure, &#39;ARG_Class&#39;), row.names = 1)</code></pre>
<pre><code>##                     Proportion
## Tetracycline             0.571
## Multiple Resistance      0.186
## Aminoglycoside           0.184
## Nucleoside               0.022
## Lincosamide              0.015
## Sulfonamide              0.011
## Streptogramin            0.005
## Macrolide                0.003
## Phenicol                 0.001
## Unclassified             0.001
## Rifamycin                0.001
## Diaminopyrimidine        0.000
## Beta-Lactam              0.000
## Aminocoumarin            0.000
## Glycopeptide             0.000
## Fluoroquinolone          0.000
## Oxazolidinone            0.000
## Pleuromutilin            0.000
## Peptide                  0.000
## Undefined                0.000</code></pre>
<pre class="r"><code>data.frame(taxa_proportions(soil, &#39;ARG_Class&#39;), row.names = 1)</code></pre>
<pre><code>##                                Proportion
## Multiple Resistance                 0.653
## Aminocoumarin                       0.076
## Aminoglycoside                      0.053
## Peptide                             0.025
## Rifamycin                           0.057
## Macrolide                           0.061
## Unclassified                        0.039
## Phenicol                            0.013
## Beta-Lactam                         0.008
## Undefined                           0.003
## Glycopeptide                        0.004
## Tetracycline                        0.005
## Antibacterial Free Fatty Acids      0.001
## Nucleoside                          0.000
## Fluoroquinolone                     0.000
## Sulfonamide                         0.000
## Diaminopyrimidine                   0.000
## Lincosamide                         0.000
## Elfamycin                           0.000
## Pleuromutilin                       0.000
## Streptogramin                       0.000</code></pre>
<pre class="r"><code>sum(taxa_sums(taxa_filter(phylo_object, treatment = treatments, subset = &#39;Manure&#39;)))/sum(taxa_sums(taxa_filter(phylo_object, treatment = treatments, subset = &#39;Soil&#39;)))</code></pre>
<pre><code>## [1] 290.9389</code></pre>
</div>
</div>
<div id="phylogeny-profiles" class="section level1">
<h1>Phylogeny Profiles</h1>
<pre class="r"><code>g &lt;- phylogeny_profile(conglomerate_samples(phylo_object, treatment = treatments, merge_on = &#39;Day&#39;), &#39;ARG_Class&#39;, treatments, relative_abundance = TRUE) + 
  labs(fill = &#39;Associated Anti-Microbial\nResistance Drug Class&#39;, x = &#39;Sampling Day&#39;) + 
  guides(fill=guide_legend(ncol=1))
g</code></pre>
<p><img src="index_files/figure-html/phylogeny-bar-plots-1.png" width="1536" /></p>
</div>
<div id="ordination" class="section level1">
<h1>Ordination</h1>
<p><img src="index_files/figure-html/tsne-relative-abundance-1.png" width="1536" /> <img src="index_files/figure-html/tsne-relative-abundance-plot-1.png" width="1536" /></p>
</div>
<div id="statistical-tests" class="section level1">
<h1>Statistical Tests</h1>
<pre class="r"><code>wilcox.test(as(manure@otu_table, &#39;matrix&#39;), as(soil@otu_table, &#39;matrix&#39;))</code></pre>
<pre><code>## 
##  Wilcoxon rank sum test with continuity correction
## 
## data:  as(manure@otu_table, &quot;matrix&quot;) and as(soil@otu_table, &quot;matrix&quot;)
## W = 1497887, p-value &lt; 2.2e-16
## alternative hypothesis: true location shift is not equal to 0</code></pre>
<pre class="r"><code>wilcox.test(as(taxa_filter(phylo_object, treatments, subset = &#39;Manure&#39;)@otu_table, &#39;matrix&#39;), 
            as(taxa_filter(phylo_object, treatments, subset = &#39;Manure-Treated Soil&#39;)@otu_table, &#39;matrix&#39;))</code></pre>
<pre><code>## 
##  Wilcoxon rank sum test with continuity correction
## 
## data:  as(taxa_filter(phylo_object, treatments, subset = &quot;Manure&quot;)@otu_table,  and as(taxa_filter(phylo_object, treatments, subset = &quot;Manure-Treated Soil&quot;)@otu_table,     &quot;matrix&quot;) and     &quot;matrix&quot;)
## W = 1395243, p-value &lt; 2.2e-16
## alternative hypothesis: true location shift is not equal to 0</code></pre>
<pre class="r"><code>wilcox.test(as(taxa_filter(phylo_object, treatments, subset = &#39;Soil&#39;)@otu_table, &#39;matrix&#39;), 
            as(taxa_filter(phylo_object, treatments, subset = &#39;Manure-Treated Soil&#39;)@otu_table, &#39;matrix&#39;))</code></pre>
<pre><code>## 
##  Wilcoxon rank sum test with continuity correction
## 
## data:  as(taxa_filter(phylo_object, treatments, subset = &quot;Soil&quot;)@otu_table,  and as(taxa_filter(phylo_object, treatments, subset = &quot;Manure-Treated Soil&quot;)@otu_table,     &quot;matrix&quot;) and     &quot;matrix&quot;)
## W = 519556, p-value = 3.53e-07
## alternative hypothesis: true location shift is not equal to 0</code></pre>
<pre class="r"><code>wilcox.test(as(taxa_filter(phylo_object, treatments, subset = &#39;Soil&#39;)@otu_table, &#39;matrix&#39;), 
            as(taxa_filter(phylo_object, treatments, subset = &#39;Manure-Treated Effluent&#39;)@otu_table, &#39;matrix&#39;))</code></pre>
<pre><code>## 
##  Wilcoxon rank sum test with continuity correction
## 
## data:  as(taxa_filter(phylo_object, treatments, subset = &quot;Soil&quot;)@otu_table,  and as(taxa_filter(phylo_object, treatments, subset = &quot;Manure-Treated Effluent&quot;)@otu_table,     &quot;matrix&quot;) and     &quot;matrix&quot;)
## W = 1384318, p-value = 0.9732
## alternative hypothesis: true location shift is not equal to 0</code></pre>
<pre class="r"><code>wilcox.test(as(taxa_filter(phylo_object, treatments, subset = &#39;Effluent&#39;)@otu_table, &#39;matrix&#39;), 
            as(taxa_filter(phylo_object, treatments, subset = &#39;Manure-Treated Effluent&#39;)@otu_table, &#39;matrix&#39;))</code></pre>
<pre><code>## 
##  Wilcoxon rank sum test with continuity correction
## 
## data:  as(taxa_filter(phylo_object, treatments, subset = &quot;Effluent&quot;)@otu_table,  and as(taxa_filter(phylo_object, treatments, subset = &quot;Manure-Treated Effluent&quot;)@otu_table,     &quot;matrix&quot;) and     &quot;matrix&quot;)
## W = 6118106, p-value = 0.1742
## alternative hypothesis: true location shift is not equal to 0</code></pre>
<pre class="r"><code>dat &lt;- taxa_filter(phylo_object, treatment = treatments, subset = &#39;Manure&#39;)
adonis(t(dat@otu_table) ~ Sample, data = as(dat@sam_data, &#39;data.frame&#39;),  permutations=1000, method = &quot;bray&quot;)</code></pre>
<pre><code>## &#39;nperm&#39; &gt;= set of all permutations: complete enumeration.</code></pre>
<pre><code>## Set of permutations &lt; &#39;minperm&#39;. Generating entire set.</code></pre>
<pre><code>## 
## Call:
## adonis(formula = t(dat@otu_table) ~ Sample, data = as(dat@sam_data,      &quot;data.frame&quot;), permutations = 1000, method = &quot;bray&quot;) 
## 
## Permutation: free
## Number of permutations: 23
## 
## Terms added sequentially (first to last)
## 
##           Df SumsOfSqs MeanSqs F.Model R2 Pr(&gt;F)
## Sample     3  0.010968       0       0  1      1
## Residuals  0  0.000000     Inf          0       
## Total      3  0.010968                  1</code></pre>
<pre class="r"><code>dat &lt;- taxa_filter(phylo_object, treatment = treatments, subset = &#39;Soil&#39;)
adonis(t(dat@otu_table) ~ Sample, data = as(dat@sam_data, &#39;data.frame&#39;),  permutations=1000, method = &quot;bray&quot;)</code></pre>
<pre><code>## &#39;nperm&#39; &gt;= set of all permutations: complete enumeration.
## Set of permutations &lt; &#39;minperm&#39;. Generating entire set.</code></pre>
<pre><code>## 
## Call:
## adonis(formula = t(dat@otu_table) ~ Sample, data = as(dat@sam_data,      &quot;data.frame&quot;), permutations = 1000, method = &quot;bray&quot;) 
## 
## Permutation: free
## Number of permutations: 23
## 
## Terms added sequentially (first to last)
## 
##           Df SumsOfSqs MeanSqs F.Model R2 Pr(&gt;F)
## Sample     3   0.12541       0       0  1      1
## Residuals  0   0.00000     Inf          0       
## Total      3   0.12541                  1</code></pre>
<pre class="r"><code>dat &lt;- taxa_filter(phylo_object, treatment = treatments, subset = &#39;Effluent&#39;)
adonis(t(dat@otu_table) ~ Day * Sample, data = as(dat@sam_data, &#39;data.frame&#39;),  permutations=1000, method = &quot;bray&quot;)</code></pre>
<pre><code>## 
## Call:
## adonis(formula = t(dat@otu_table) ~ Day * Sample, data = as(dat@sam_data,      &quot;data.frame&quot;), permutations = 1000, method = &quot;bray&quot;) 
## 
## Permutation: free
## Number of permutations: 1000
## 
## Terms added sequentially (first to last)
## 
##           Df SumsOfSqs MeanSqs F.Model      R2 Pr(&gt;F)
## Day        2   0.70022       0       0 0.51473      1
## Sample     9   0.66014       0       0 0.48527      1
## Residuals  0   0.00000     Inf         0.00000       
## Total     11   1.36036                 1.00000</code></pre>
<pre class="r"><code>dat &lt;- taxa_filter(phylo_object, treatment = treatments, subset = &#39;Manure-Treated Soil&#39;)
adonis(t(dat@otu_table) ~ Day * Sample, data = as(dat@sam_data, &#39;data.frame&#39;),  permutations=1000, method = &quot;bray&quot;)</code></pre>
<pre><code>## 
## Call:
## adonis(formula = t(dat@otu_table) ~ Day * Sample, data = as(dat@sam_data,      &quot;data.frame&quot;), permutations = 1000, method = &quot;bray&quot;) 
## 
## Permutation: free
## Number of permutations: 1000
## 
## Terms added sequentially (first to last)
## 
##           Df SumsOfSqs MeanSqs F.Model      R2 Pr(&gt;F)
## Day        2   0.39011       0       0 0.15942      1
## Sample     9   2.05687       0       0 0.84058      1
## Residuals  0   0.00000     Inf         0.00000       
## Total     11   2.44698                 1.00000</code></pre>
<pre class="r"><code>dat &lt;- taxa_filter(phylo_object, treatment = treatments, subset = &#39;Manure-Treated Effluent&#39;)
adonis(t(dat@otu_table) ~ Day * Sample, data = as(dat@sam_data, &#39;data.frame&#39;),  permutations=1000, method = &quot;bray&quot;)</code></pre>
<pre><code>## 
## Call:
## adonis(formula = t(dat@otu_table) ~ Day * Sample, data = as(dat@sam_data,      &quot;data.frame&quot;), permutations = 1000, method = &quot;bray&quot;) 
## 
## Permutation: free
## Number of permutations: 1000
## 
## Terms added sequentially (first to last)
## 
##           Df SumsOfSqs MeanSqs F.Model      R2 Pr(&gt;F)
## Day        5    2.4338       0       0 0.69899      1
## Sample    18    1.0481       0       0 0.30101      1
## Residuals  0    0.0000    -Inf         0.00000       
## Total     23    3.4818                 1.00000</code></pre>
</div>
<div id="identifying-persisting-genes-from-manure" class="section level1">
<h1>Identifying Persisting Genes from Manure</h1>
<pre class="r"><code>sum(unique_control_genes$Manure %in% taxa_names(taxa_filter(phylo_object, &#39;Treatment&#39;, &#39;Manure&#39;)))</code></pre>
<pre><code>## [1] 75</code></pre>
<pre class="r"><code>sum(unique_control_genes$Manure %in% taxa_names(taxa_filter(phylo_object, treatments, &#39;Manure-Treated Effluent&#39;)))</code></pre>
<pre><code>## [1] 69</code></pre>
<pre class="r"><code>manure_persisters_genes &lt;- unique_control_genes$Manure[unique_control_genes$Manure %in% taxa_names(taxa_filter(phylo_object, &#39;Treatment&#39;, &#39;Manure&#39;, frequency = 0.5))]
persisters_in_manure &lt;- taxa_filter(subset_taxa(phylo_object, taxa_names(phylo_object) %in% manure_persisters_genes), treatment = treatments, subset = &#39;Manure&#39;, drop_samples = TRUE)
sum(taxa_sums(persisters_in_manure)) / sum(taxa_sums(taxa_filter(phylo_object, treatments, subset = &#39;Manure&#39;)))</code></pre>
<pre><code>## [1] 0.162051</code></pre>
<pre class="r"><code>manure_persisters &lt;- taxa_filter(subset_taxa(phylo_object, taxa_names(phylo_object) %in% manure_persisters_genes), treatment = &#39;Treatment&#39;, subset = &#39;Manure&#39;, drop_samples = TRUE)</code></pre>
<pre class="r"><code>manure_persisters &lt;- conglomerate_samples(manure_persisters, treatments, merge_on = &#39;Day&#39;)</code></pre>
<pre class="r"><code>data.frame(taxa_proportions(subset_taxa(manure_persisters, &#39;Matrix_Treatment&#39; != &#39;Manure&#39;), &#39;ARG_Class&#39;, treatments))</code></pre>
<pre><code>##           Matrix_Treatment           ARG_Class Proportion
## 1      Manure-Treated Soil      Aminoglycoside      0.389
## 2      Manure-Treated Soil        Tetracycline      0.191
## 3      Manure-Treated Soil Multiple Resistance      0.188
## 4      Manure-Treated Soil         Lincosamide      0.144
## 5      Manure-Treated Soil         Beta-Lactam      0.082
## 6      Manure-Treated Soil            Phenicol      0.006
## 7  Manure-Treated Effluent      Aminoglycoside      0.355
## 8  Manure-Treated Effluent        Tetracycline      0.314
## 9  Manure-Treated Effluent         Lincosamide      0.149
## 10 Manure-Treated Effluent Multiple Resistance      0.132
## 11 Manure-Treated Effluent         Beta-Lactam      0.042
## 12 Manure-Treated Effluent            Phenicol      0.009</code></pre>
<pre class="r"><code>phylogeny_profile(manure_persisters, classification = &#39;ARG_Class&#39;, treatment = treatments, relative_abundance = FALSE) + 
  labs(color = &#39;Associated Anti-Microbial\nResistance Drug Class&#39;, fill = &#39;Associated Anti-Microbial\nResistance Drug Class&#39;, x = &#39;Sampling Day&#39;)</code></pre>
<p><img src="index_files/figure-html/manure-persister-composition-profile-1.png" width="1536" /></p>
<pre class="r"><code>graph_data &lt;- melt_phyloseq(manure_persisters)
colors &lt;- phylosmith:::create_palette(length(unique(graph_data[[&#39;ARG_Class&#39;]])), &#39;default&#39;)
abundance_heatmap(manure_persisters, classification = &#39;Gene&#39;, treatment = treatments, transformation = &#39;log&#39;) + 
  theme(axis.text.y = element_text(colour = colors, face = &#39;bold&#39;)) + labs(color = &#39;Natural Log of\nGene Abundance&#39;, fill = &#39;Natural Log of\nGene Abundance&#39;, x = &#39;Sampling Day&#39;)</code></pre>
<p><img src="index_files/figure-html/all-persister-heatmaps-merged-1.png" width="1536" /></p>
<pre class="r"><code>abundance_lines(manure_persisters, &#39;Gene&#39;, treatments) + scale_x_discrete(expand = expand_scale(mult = c(0,0.02))) + 
  theme(panel.spacing = unit(3, &quot;cm&quot;)) + guides(colour = guide_legend(ncol = 1, override.aes = list(size = 4))) + 
  labs(color = &#39;Gene&#39;, x = &#39;Sampling Day&#39;)</code></pre>
<pre><code>## Scale for &#39;x&#39; is already present. Adding another scale for &#39;x&#39;, which
## will replace the existing scale.</code></pre>
<p><img src="index_files/figure-html/all-persister-lines-1.png" width="1536" /></p>
</div>
<div id="identify-enriched-genes" class="section level1">
<h1>Identify Enriched Genes</h1>
<pre class="r"><code>Soil &lt;- taxa_filter(phylo_object, treatments, subset = &#39;Soil&#39;, frequency = 0)
Soil_genes &lt;- taxa_names(Soil)
effluent &lt;- taxa_filter(phylo_object, treatments, subset = &#39;Effluent&#39;, frequency = 0)
effluent_genes &lt;- taxa_names(effluent)
manure_treated_soil &lt;- taxa_filter(phylo_object, treatments, subset = &#39;Manure-Treated Soil&#39;)
manure_treated_soil_genes &lt;- taxa_names(manure_treated_soil)

water_stimulated &lt;- effluent_genes[!(effluent_genes %in% Soil_genes)]
water_stimulated &lt;- water_stimulated[!(water_stimulated %in% manure_genes)]; length(water_stimulated)</code></pre>
<pre><code>## [1] 54</code></pre>
<pre class="r"><code>water_enhanced &lt;- effluent_genes[effluent_genes %in% Soil_genes]
water_enhanced &lt;- water_enhanced[taxa_sums(subset_taxa(effluent, taxa_names(effluent) %in% water_enhanced))/nsamples(effluent) &gt; taxa_sums(subset_taxa(Soil, taxa_names(Soil) %in% water_enhanced))/nsamples(Soil)]; length(water_enhanced)</code></pre>
<pre><code>## [1] 44</code></pre>
<pre class="r"><code>water_enriched &lt;- c(water_enhanced, water_stimulated); length(water_enriched)</code></pre>
<pre><code>## [1] 98</code></pre>
<pre class="r"><code>manure_stimulated &lt;- manure_treated_soil_genes[!(manure_treated_soil_genes %in% soil_genes)]
manure_stimulated &lt;- manure_stimulated[!(manure_stimulated %in% manure_genes)]
length(manure_stimulated)</code></pre>
<pre><code>## [1] 31</code></pre>
<pre class="r"><code>manure_enhanced &lt;- manure_treated_soil_genes[manure_treated_soil_genes %in% soil_genes]
manure_enhanced &lt;- manure_enhanced[taxa_sums(subset_taxa(manure_treated_soil, taxa_names(manure_treated_soil) %in% manure_enhanced))/nsamples(manure_treated_soil) &gt; taxa_sums(subset_taxa(soil, taxa_names(soil) %in% manure_enhanced))/nsamples(soil)]
manure_enhanced &lt;- manure_enhanced[!(manure_enhanced %in% water_enriched)]; length(manure_enhanced)</code></pre>
<pre><code>## [1] 63</code></pre>
<pre class="r"><code>new &lt;- as(phylo_object@tax_table, &#39;matrix&#39;)
new &lt;- new[,c(6,4)]
newnew &lt;- phylo_object
newnew@tax_table &lt;- tax_table(new)
newnew &lt;- conglomerate_samples(newnew, treatment = treatments)
new &lt;- conglomerate_taxa(newnew, &#39;ARG_Class&#39;)

graph_data &lt;- melt_phyloseq(taxa_filter(new, treatment = &#39;Matrix_Treatment&#39;, subset = &#39;Manure-Treated Effluent&#39;))
graph_data$Classification &lt;- factor(graph_data$Classification, levels = c(&#39;Water_Enriched&#39;, &#39;Manure_Enriched&#39;, &#39;Manure_Persister&#39;, &#39;Soil&#39;, &#39;Background&#39;))
levels(graph_data$Classification) &lt;- c(&#39;Water Enriched&#39;, &#39;Manure Enriched&#39;, &#39;Manure Persister&#39;, &#39;Soil&#39;, &#39;Background&#39;)
# graph_data[, Relative_Abundance := Abundance/sum(Abundance), by = Classification]
graph_data &lt;- graph_data[!(graph_data$Classification %in% c(&#39;Soil&#39;, &#39;Background&#39;)), ]
color_count &lt;- length(unique(graph_data[[&#39;ARG_Class&#39;]]))
graph_colors &lt;- phylosmith:::create_palette(color_count)


ggplot(graph_data, aes_string(x = &quot;Classification&quot;, y = &quot;Abundance&quot;, 
                                   fill = &#39;ARG_Class&#39;)) + 
  guides(fill = guide_legend(ncol = ceiling(length(unique(graph_data[[&#39;ARG_Class&#39;]]))/25)), color = FALSE) + 
  scale_fill_manual(values = graph_colors, aesthetics = c(&quot;color&quot;, &quot;fill&quot;)) + 
  facet_grid(reformulate(&#39;Classification&#39;), scales = &quot;free&quot;, space = &quot;free&quot;) + 
  geom_bar(aes_string(color = &#39;ARG_Class&#39;, fill = &#39;ARG_Class&#39;), stat = &quot;identity&quot;, position = &quot;stack&quot;, size = 0.8, color = &#39;black&#39;) + 
  ylab(&#39;Abundance&#39;) + 
  theme_classic() + theme(text = element_text(family = &quot;times&quot;),
                                axis.text.x = element_blank(), axis.text.y = element_text(size = 10), axis.title.x = element_blank(), axis.title.y = element_text(size = 10, 
                                                                                                                                                            face = &quot;bold&quot;), axis.ticks.x = element_blank(), 
                          legend.title = element_text(size = 10, face = &quot;bold&quot;), 
                          legend.text = element_text(size = 8), 
                          legend.spacing.x = unit(0.01, &quot;npc&quot;), 
                          panel.background = element_rect(color = &quot;black&quot;, 
                                                          size = 1.5, fill = &quot;white&quot;), panel.spacing = unit(0.01, &quot;npc&quot;), strip.text.x = element_text(size = 10, face = &quot;bold&quot;), 
                          strip.background = element_rect(colour = &quot;black&quot;, size = 1.4),
                          legend.key.size = unit(4, &quot;mm&quot;)) + 
  scale_y_continuous(expand = expand_scale(mult = c(0.0037, 0.003), add = c(0, 0))) + 
  scale_x_discrete(expand = expand_scale(mult = 0, add = 0.51)) + 
labs(color = &#39;Associated Anti-Microbial\nResistance Drug Class&#39;, fill = &#39;Associated Anti-Microbial\nResistance Drug Class&#39;, x = &#39;Sampling Day&#39;)</code></pre>
<p><img src="index_files/figure-html/gene-profile-1.png" width="1536" /></p>
<pre class="r"><code>taxa_proportions(phylo_object, classification = &#39;Classification&#39;, treatment = treatments)[Matrix_Treatment == &#39;Manure-Treated Effluent&#39;]</code></pre>
<pre><code>##           Matrix_Treatment   Classification Proportion
## 1: Manure-Treated Effluent  Manure_Enriched      0.612
## 2: Manure-Treated Effluent   Water_Enriched      0.308
## 3: Manure-Treated Effluent Manure_Persister      0.071
## 4: Manure-Treated Effluent             Soil      0.003
## 5: Manure-Treated Effluent       Background      0.006</code></pre>
<pre class="r"><code>taxa_proportions(phylo_object, classification = &#39;Classification&#39;, treatment = treatments)[Matrix_Treatment == &#39;Effluent&#39;]</code></pre>
<pre><code>##    Matrix_Treatment   Classification Proportion
## 1:         Effluent   Water_Enriched      0.973
## 2:         Effluent             Soil      0.016
## 3:         Effluent  Manure_Enriched      0.011
## 4:         Effluent       Background      0.000
## 5:         Effluent Manure_Persister      0.000</code></pre>
</div>
<div id="co-occurrence" class="section level1">
<h1>Co-Occurrence</h1>
<pre class="r"><code>rho_cutoff &lt;- permute_rho(phylo_object, treatments, replicate_samples = &#39;Day&#39;, permutations = 1000)
quantile_permuted_rhos(rho_cutoff)</code></pre>
<pre class="r"><code>co_occurrence_table &lt;- co_occurrence(phylo_object, treatments, p = 0.05)
co_occurrence_table$p &lt;- p.adjust(co_occurrence_table$p, &quot;BH&quot;)
co_occurrence_table &lt;- co_occurrence_table[p &lt; .05]
co_occurrence_table &lt;- co_occurrence_table[rho &lt;= -0.68 | rho &gt;= 0.86]</code></pre>
<pre class="r"><code>layout &lt;- network_layout_ps(phylo_object, &#39;Classification&#39;, treatments, &#39;Manure-Treated Effluent&#39;, co_occurrence_table, algorithm = &#39;fr&#39;)
a &lt;- co_occurrence_network(phylo_object, &#39;Classification&#39;, treatments, &#39;Manure-Treated Effluent&#39;, co_occurrence_table, layout = layout, cluster = TRUE) + 
  labs(color = &#39;Gene Behavior&#39;, fill = &#39;Gene Behavior&#39;)

b &lt;- co_occurrence_network(phylo_object, &#39;ARG_Class&#39;, treatments, &#39;Manure-Treated Effluent&#39;, co_occurrence_table, layout = layout, cluster = TRUE) + 
  labs(color = &#39;Associated Anti-Microbial\nResistance Drug Class&#39;, fill = &#39;Associated Anti-Microbial\nResistance Drug Class&#39;)

grid.arrange(b, a, ncol = 2)</code></pre>
<p><img src="index_files/figure-html/effluent_networks_gene-1.png" width="1536" /></p>
<pre class="r"><code>co_occurrence_network(phylo_object, &#39;ARG_Mechanism&#39;, treatments, &#39;Manure-Treated Effluent&#39;, co_occurrence_table, layout = layout, cluster = TRUE, nodes_of_interest = c(&#39;tet(33)&#39;, &#39;ermB&#39;))</code></pre>
<p><img src="index_files/figure-html/effluent_networks_mechanism-1.png" width="1536" /></p>
</div>


<p><br>
<br>
<strong><a href="https://schuyler-smith.github.io/" target="_blank">Schuyler D. Smith</a></strong>
<br>
Ph.D. Student - Bioinformatics and Computational Biology<br>
Iowa State University.  Ames, IA.</p>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = false;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
